File: Sorting and Searching/Playlist.java
import java.io.*;
import java.util.*;
public class Playlist{
    public static void main(String args[])throws IOException
    {
    InputStreamReader r=new InputStreamReader(System.in);    
    Scanner sc=new Scanner(System.in);
    int n=sc.nextInt();
    long nArray[]=new long[n];
    for(int i=0;i<n;i++) 
        nArray[i]=sc.nextLong();
    int start=0;
    int length=1;
    int max=Integer.MIN_VALUE;
    HashMap<Long,Integer> pos=new HashMap<>();
    pos.put(nArray[0],0);
    for(int i=1; i<n; i++){
        //System.out.println(length);
        if(pos.containsKey(nArray[i])){
            start=Math.max(start,pos.get(nArray[i])+1);
            length=(i-start)+1;
            pos.replace(nArray[i],i);
        } 
        else
        {
            pos.put(nArray[i],i);
            length++;
        }
 
        if(max<length)
            max=length;
 
    }
    System.out.println(max);
}
}

File: Sorting and Searching/Playlist.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n,x;
    cin>>n;
    vector<int> nArray(n);
    for(int i=0;i<n;i++){
        cin>>nArray[i];
    }
    int start=0;
    int length=1;
    int max=INT_MIN;
    std::map<int, int> pos;
    pos[nArray[0]]=0;
    for(int i=1; i<n; i++){
        //cout<<length<<'\n';
        //System.out.println(length);
        if(pos.find(nArray[i]) != pos.end()){
            start=std::max(start,pos[nArray[i]]+1);
            length=(i-start)+1;
            pos[nArray[i]]=i;
        } 
        else
        {
            pos[nArray[i]]=i;
            length++;
        }
        max=std::max(max,length);

 
    }
    cout<<max;
    return 0;
}